rdrf/rdrf/features/steps.py

```diff
@@ -41,9 +41,58 @@ def scroll_to(element):
      scroll_to_y(y)
      print("scrolled to %s at (0,Y) = %s" % (moniker(element), y))
      return y
+
+
+def scroll_to_multisection_cde(section, cde, item=1):
+    # item 1 means the 1st block of cdes in the multisection
+    print("Attempting to scroll to section %s cde %s item %s" % (section,
+                                                                 cde,
+                                                                 item))
+    
+    formset_string = "-%s-" % (int(item) - 1)
+    print("formset_string = %s" % formset_string)
+    if item == 1:
+        # the first (default) item is not created by js from the empty form template
+        contains_clause = " not(contains(,, '__prefix__')) "
+    else:
+        # subsequent items are ...
+        contains_clause = " contains(., '__prefix__') "
+
+    print("contains clause = %s" % contains_clause)
+    xpath = "//div[@class='panel-heading' and contains(., '%s')]" % section
+    default_panel = world.browser.find_element_by_xpath(xpath).find_element_by_xpath("..")
+    
+    label_expression = ".//label[contains(., '%s')]" % cde
+    
+    for label_element in default_panel.find_elements_by_xpath(label_expression):
+        print("found a label element for cde %s" % cde)
+        input_div = label_element.find_element_by_xpath(".//following-sibling::div")
+        try:
+            input_element = input_div.find_element_by_xpath(".//input[contains(@id, '%s')]" % formset_string)
+        except:
+            input_element = None
+            
+        if not input_element:
```

Comment:
Minor point but I would flip this conditional. ie the norm is to continue the loop and the exception is to return.